cobalt/black_box_tests/testdata/scroll.html

```diff
@@ -35,18 +34,109 @@
 <body>
   <div class="app" style="overflow:auto;">
     <div id="row" class="row" style="height: 20rem;">
-      <div id="top_one" class="tile">One</div>
+      <div id="tile" class="tile">One</div>
     </div>
   </div>
 </body>
 <script>
+  // Fail if the test is not finished within 15 seconds.
+  var kTimeout = 30 * 1000;
+  var failTimer = setTimeout(fail, kTimeout);
+
+  function fail() {
+    console.log("Failing due to timeout!");
+    assertTrue(false);
+  }
+
+  function phasename(phase) {
+    switch (phase) {
+      case 0: return 'none';
+      case 1: return 'capturing';
+      case 2: return 'at target';
+      case 3: return 'bubbling';
+    }
+    return ' [unknown: ' + phase + ']';
+  }
+
+  const expected_events = [
+    // name, id, phase
+    // actions.move_to_element(row).pause(_SLEEP_AFTER_MOVE_TIME)
+    ['pointermove', 'tile', 'at target'],
+    ['pointermove', 'row', 'bubbling'],
+
+    // actions.click_and_hold(row).pause(_SLEEP_AFTER_MOVE_TIME)
+    ['pointermove', 'tile', 'at target'],
+    ['pointermove', 'row', 'bubbling'],
+
+    // actions.move_by_offset(-500, 0).pause(_SLEEP_AFTER_MOVE_TIME)
+    ['pointerdown', 'tile', 'at target'],
+    ['pointerdown', 'row', 'bubbling'],
+    ['pointercancel', 'tile', 'at target'],
+    ['pointercancel', 'row', 'bubbling'],
+    ['scroll', 'row', 'at target'],
+  ]
+
+  let current_event_number = 0
+  let failure_count = 0;
+
+  function CheckEvent(name, id, phase) {
+    while ((current_event_number < expected_events.length) &&
```

Comment:
Do you think this is better than failing fast? I'd be inclined to just have it stop the test the first time it gets a failure, but both have merits

Reply:
When writing the test, I feel the existing way might be a little bit better. By not failing the test immediately, we can see what are the events missing and what are the unexpected events. I think it would be useful for debugging as well if this test fails in build bot and we can immediately know what's wrong, so I prefer to just let it log all the events.

Reply:
Sgtm